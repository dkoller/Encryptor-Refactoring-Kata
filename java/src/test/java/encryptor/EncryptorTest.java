package encryptor;

import org.approvaltests.Approvals;
import org.junit.jupiter.api.Test;

import java.io.ByteArrayOutputStream;
import java.io.PrintStream;
import java.io.UnsupportedEncodingException;
import java.security.InvalidParameterException;

import static org.junit.jupiter.api.Assertions.assertThrows;
import static org.junit.jupiter.api.Assertions.fail;

public class EncryptorTest {

    @Test
    public void cryptSentence() {
      Approvals.verify(new Encryptor().cryptSentence("coding is like cooking"));
    }


    @Test
    public void printWords() throws UnsupportedEncodingException {
        ByteArrayOutputStream baos = new ByteArrayOutputStream();
        PrintStream ps = new PrintStream(baos,true,"UTF-8");
        new Encryptor().printWords("coding is like cooking", ps);
        String output = baos.toString("UTF-8");
        Approvals.verify(output);
    }

    @Test
    public void cryptWord() {
        Approvals.verify(new Encryptor().cryptWord("cooking"));
    }

    @Test
    public void cryptWordWithSpaces() {
        assertThrows(IllegalArgumentException.class, () -> new Encryptor().cryptWord("coding is like cooking"));
    }

    @Test
    public void cryptWordToNumbers() {
        Approvals.verify(new Encryptor().cryptWordToNumbers("coding"));
    }


    @Test
    public void cryptWordToNumbersWithSpaces() {
        assertThrows(IllegalArgumentException.class, () -> new Encryptor().cryptWordToNumbers("coding  is like cooking"));
    }

    @Test
    public void cryptWordAndReplaceWithSpaces() {
        assertThrows(InvalidParameterException.class, () -> new Encryptor().cryptWord("coding is like cooking", ""));
    }

    @Test
    public void cryptWordAndReplace() {
        Approvals.verify(new Encryptor().cryptWord("coding", "cooking"));
    }
}
