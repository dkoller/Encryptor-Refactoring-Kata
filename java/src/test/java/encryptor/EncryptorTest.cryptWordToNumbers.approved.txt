101113102107112105